Introduce a pass-through transaction option setter

behos opened the pull request:
This allows passing through code and bytes directly to set options, bypassing the generated option types. This is handy when receiving already serialized code and data to not require mapping it to the enum and back to raw data.

PierreZ commented:
Hi @behos :wave: 

Thanks for opening the PR :smile: 

Could you add a small, dummy test about `set_raw_option`? :pray:

behos commented:
> Hi @behos wave
> 
> Thanks for opening the PR smile
> 
> Could you add a small, dummy test about `set_raw_option`? pray

Yes, of course! It was an oversight not adding one in the first place.

Added the test and fixed a bug that it detected :)

behos commented:
Hi @PierreZ , it looks like the checks are passing successfully :) Is there something else to change in this PR or should I leave it to you to merge when convenient?

PierreZ commented:
Thanks for your contribution :rocket:
